src/test/java/teammates/test/cases/action/AdminEmailComposeSaveActionTest.java

```diff
@@ -253,27 +254,34 @@ public void testExecuteAndPostProcess() throws Exception {
         assertEquals(receiver, StringHelper.join(", ", savedEmail.getAddressReceiver().toArray(new String[0])));
 
         ______TS("save non-existing email : invalid subject : failure");
-        emailId = "nonExisitingId";
-        content = "valid content";
-        subject = "";
-        receiver = "test@example.tmt";
-        action = getAction(
-                Const.ParamsNames.ADMIN_EMAIL_CONTENT, content,
-                Const.ParamsNames.ADMIN_EMAIL_SUBJECT, subject,
-                Const.ParamsNames.ADMIN_EMAIL_ADDRESS_RECEIVERS, receiver,
-                Const.ParamsNames.ADMIN_EMAIL_ID, emailId);
-        pageResult = getShowPageResult(action);
-        assertEquals(getPageResultDestination(Const.ViewURIs.ADMIN_EMAIL, true, "admin.user"),
-                pageResult.getDestinationWithParams());
-
-        expectedLogSegment = Const.ACTION_RESULT_FAILURE;
-        AssertHelper.assertContains(expectedLogSegment, action.getLogMessage());
 
-        expectedStatus = "The field 'email subject' is empty.";
-        AssertHelper.assertContains(expectedStatus, pageResult.getStatusMessage());
-
-        data = (AdminEmailComposePageData) pageResult.data;
-        assertEquals(subject, data.emailToEdit.subject);
+        try {
+            emailId = "nonExisitingId";
+            content = "valid content";
+            subject = "";
+            receiver = "test@example.tmt";
+            action = getAction(
+                    Const.ParamsNames.ADMIN_EMAIL_CONTENT, content,
+                    Const.ParamsNames.ADMIN_EMAIL_SUBJECT, subject,
+                    Const.ParamsNames.ADMIN_EMAIL_ADDRESS_RECEIVERS, receiver,
+                    Const.ParamsNames.ADMIN_EMAIL_ID, emailId);
```

Comment:
It seems that exception will occur in that line and all lines above will not executed. I think its better to not add AssertNotEmpty to builder constructor since we have other mechanism to handle such cases.